openshift/release/common.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+# Collection of shared functions used to generate and prepare releases.
+
+ROOT_DIR=$(dirname $0)/../..
+
+readonly FAAS_VERSION=${FAAS_VERSION:-"main"}
+readonly FAAS_REPO=${FAAS_REPO:-"github.com/boson-project/faas"}
+
+# The vendor/ dir is omitted to be added separately
+UPDATED_FILES=$(cat <<EOT | tr '\n' ' '
+pkg/kn/root/plugin_register.go
+go.mod
+go.sum
+EOT
+)
+
+# Generates plugin_register.go file to enable plugin inlining
+generate_file() {
+  echo ":: Generating plugin_register.go file ::"
+  local faas_repo=$1
+  
+  cat <<EOF > "${ROOT_DIR}/pkg/kn/root/plugin_register.go"
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package root
+
+import (
+	_ "${faas_repo}/plugin"
+)
+
+// RegisterInlinePlugins is an empty function which however forces the
+// compiler to run all init() methods of the registered imports
+func RegisterInlinePlugins() {}
+EOF
+}
+
```

Comment:
I also think, that having the go file as a separate file (not inlined) would make it easier to maintain/edit according to the formatting options.

When we probably add more plugins it's then also easier to maintain, so I would create:


* A skeleton `.go` file that has just an empty import list and with comment markers. 
* During the update, that go file would be copied over to it final destination and during the copy the imports where added by replacing the comment marker with a list of plugins (currently its only a single plugin, but we could have just a list of imports to add)
* After the replacement we then call `gofmt` to clean up according to golang rules

If this is easy to implement, we could do that right now but we also can do it just in a separate PR. It's up to you.

Reply:
I've seen that coming. :smiley: Ok, so let's make it properly, I'll update the PR shortly.

Reply:
Updated in https://github.com/openshift/knative-client/pull/493/commits/d2ff8bdc4e21b03c723129ba518c3268d411116a